pint_xarray/accessors.py

```diff
@@ -230,15 +252,109 @@ def registry(self):
+        Dimensions without coordinates: x
+
+        Convert the data
+
+        >>> da.pint.to("mm")
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) int64 <Quantity([0 1 2 3 4], 'second')>
+        Dimensions without coordinates: x
+        >>> da.pint.to(ureg.mm)
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) int64 <Quantity([0 1 2 3 4], 'second')>
+        Dimensions without coordinates: x
+        >>> da.pint.to({da.name: "mm"})
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) int64 <Quantity([0 1 2 3 4], 'second')>
+        Dimensions without coordinates: x
+
+        Convert coordinates
+
+        >>> da.pint.to({"u": ureg.ms})
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([0.   0.25 0.5  0.75 1.  ], 'meter')>
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        >>> da.pint.to(u="ms")
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([0.   0.25 0.5  0.75 1.  ], 'meter')>
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+
+        Convert both simultaneously
+
+        >>> da.pint.to("mm", u="ms")
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        >>> da.pint.to({"arr": ureg.mm, "u": ureg.ms})
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        >>> da.pint.to(arr="mm", u="ms")
+        <xarray.DataArray 'arr' (x: 5)>
+        <Quantity([   0.  250.  500.  750. 1000.], 'millimeter')>
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        """
+        if isinstance(units, (str, pint.Unit)):
+            unit_kwargs[self.da.name] = units
+            units = None
+        elif units is not None and not is_dict_like(units):
+            raise ValueError(
+                "units must be either a string, a pint.Unit object or a dict-like,"
+                f" but got {units!r}"
+            )
+
+        units = either_dict_or_kwargs(units, unit_kwargs, "to")
+
+        return conversion.convert_units(self.da, units)
 
     def to_base_units(self):
```

Comment:
Should `to_base_units` also be adjusted analogously to use a util from `conversion` or is it fine as-is?

Reply:
`to_base_units` uses `Quantity.to_base_units`, so I'm not quite sure how we'd be able to use a `conversion` function here. Did you want to add a function to `conversion` that calls `.to_base_units`? If so, then yes, we could use that to reduce duplicated code.

We should also allow passing in a `dict` / kwargs to control which of the `DataArray`'s variables should be converted (defaulting to "convert all variables").

Both should be a new PR, though.